xcube/core/dsio.py

```diff
@@ -556,21 +554,15 @@ def _get_path_or_store(path: str,
             client_kwargs['aws_secret_access_key'] = client_kwargs.pop('provider_secret_access_key')
 
     if path.startswith("https://") or path.startswith("http://"):
-        import urllib3.util
-        url = urllib3.util.parse_url(path_or_store)
-        if url.port is not None:
-            client_kwargs['endpoint_url'] = f'{url.scheme}://{url.host}:{url.port}'
-        else:
-            client_kwargs['endpoint_url'] = f'{url.scheme}://{url.host}'
-        root = url.path
-        if root.startswith('/'):
-            root = root[1:]
+        endpoint_url, root = split_bucket_url(path)
+        if endpoint_url:
+            client_kwargs['endpoint_url'] = endpoint_url
```

Comment:
Please also check if the code at

* https://github.com/dcs4cop/xcube/blob/master/xcube/core/mldataset.py#L610-L620

can be replaced by a function call that also simplifies read()/write() here. I imagine a function that takes our args and kwargs and outputs an s3fs.S3Map instance for read and/or write access.